### PR TITLE
Add templating example using Swig and YAML frontmatter

### PR DESCRIPTION
**Pull request by mil**
I was at a lack of documentation when trying to setup YAML front matter + templating for gulp. Here's a simple recipe for setting up gulp with YAML front matter and swig.

### CODE REVIEW

File: docs/recipes/templating-with-swig-and-yaml-front-matter.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+##### `gulpfile.js`
+
+```js
+var gulp = require('gulp');
+var swig = require('gulp-swig');
+var data = require('gulp-data');
+var fm   = require('gulp-front-matter');
+
+gulp.task("compile-page", function() {
```
Comment (sindresorhus) [inline docs/recipes/templating-with-swig-and-yaml-front-matter.md]: single-quotes

File: docs/recipes/templating-with-swig-and-yaml-front-matter.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+</html>
+```
+
+##### `gulpfile.js`
+
+```js
+var gulp = require('gulp');
+var swig = require('gulp-swig');
+var data = require('gulp-data');
+var fm   = require('gulp-front-matter');
```
Comment (sindresorhus) [inline docs/recipes/templating-with-swig-and-yaml-front-matter.md]:
```
var fm = require('gulp-front-matter');
```

File: docs/recipes/templating-with-swig-and-yaml-front-matter.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+gulp.task("compile-page", function() {
+  gulp.src('page.html')
+      .pipe(fm({
+          property : 'vars',
+          remove   : true
```
Comment (sindresorhus) [inline docs/recipes/templating-with-swig-and-yaml-front-matter.md]:
```
property: 'vars',
remove: true
```

File: docs/recipes/templating-with-swig-and-yaml-front-matter.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      }))
+      .pipe(data(function(page) { 
+          return page.vars;
+      }))
+      .pipe(swig())
+      .pipe(gulp.dest('build'));
+});
+
+gulp.task("default", ["compile-page"]);
```
Comment (sindresorhus) [inline docs/recipes/templating-with-swig-and-yaml-front-matter.md]: single-quotes

**Comment by shinnn** [comment]
gulp-data isn't needed if you set gulp-front-matter option to `property: 'data'`.

File: docs/recipes/templating-with-swig-and-yaml-front-matter.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+```js
+var gulp = require('gulp');
+var swig = require('gulp-swig');
+var frontMatter = require('gulp-front-matter');
+
+gulp.task('compile-page', function() {
+  gulp.src('page.html')
+      .pipe(frontMatter({
+          property: 'data',
+          remove: true
```
Comment (shinnn) [inline docs/recipes/templating-with-swig-and-yaml-front-matter.md]: `remove` option is `true` by default.

**Comment by phated** [comment]
Due to a bunch of unfortunate, but necessary rebases, this is out of sync.  Can you rebase your changes on the new master?  Thanks.

**Comment by mil** [comment]
Rebased and cleaned up the example a bit more.

Should be clean to be upstreamed -- let me know anything else standing in the way.

**Pull request closed without merging**